plugins/tech-insights/src/components/FactDetails/FactDetails.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
+import { useApi } from '@backstage/core-plugin-api';
+import { techInsightsApiRef } from '../../api/TechInsightsApi';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { getCompoundEntityRef } from '@backstage/catalog-model';
+import Fact from './Fact';
+import { useMemo } from 'react';
+import { List, ListItem } from '@material-ui/core';
+
+export const FactDetails = (props: {
```

Comment:
The naming of this components seems a bit confusing. It's displaying results of each fact that is confgured for a boolean check itself. 

A better approach for this is to create a new type of check that produces these kind of metric / calculation instead of doing it purely on the frontend side. You can achieve this by creating a new CheckEngine. 

The engine in this case I assume would receive a check config with the wanted fact ids and their denominator/numerator keys/identifiers, and then storing/returning the result of that check calculation.